src/notify.rs

```diff
@@ -620,3 +613,13 @@ mod __private {
     pub trait Sealed {}
     impl<N: super::NotificationPrivate + ?Sized> Sealed for N {}
 }
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    #[should_panic]
+    fn negative_notification_should_panic() {
+        let event = crate::Event::new();
+        event.notify(-1);
```

Comment:
why were negative notifications allowed at all (and not prevented by using a type that doesn't have these)?

Reply:
Because by default numeric literals passed into `notify` will resolve to `i32`, so to handle this common case we need to handle this number.